src/core/pointcloud/qgspointcloudlayer.h

```diff
@@ -246,7 +275,12 @@ class CORE_EXPORT QgsPointCloudLayer : public QgsMapLayer
 
     QgsPointCloudLayerElevationProperties *mElevationProperties = nullptr;
 
+    LayerOptions mLayerOptions;
+
     bool mSync3DRendererTo2DRenderer = false;
+    std::unique_ptr<QgsPointCloudStatsCalculator> mStatsCalculator;
```

Comment:
I don't think we need to keep the stats calculator class as a member, ideally we need just the calculated stats extracted from the calculator from the task.